### ISSUE TITLE
[Proposal][Optional][Experimental] Luhmann Social Systems analysis layer: functional systems, communication codes and cross-system translation

### ISSUE THREAD

**Issue by TomiToivio** [TomiToivio/LaclauGPT-Data-Analysis#27]
## Status

**Proposal / optional / experimental analysis layer.**

This should **not** become a mandatory part of the default LaclauGPT pipeline. Implement only behind an explicit feature/config flag and preserve all existing Laclau/Mouffe/Palonen analysis semantics.

## Motivation

Add an experimental **Niklas Luhmann / Social Systems** analysis layer for asking a different question from the existing discourse pipeline:

> In what social/functional-system context does a communication operate, what distinctions/codes does it mobilize, and how is meaning translated when communication moves between systems such as science, politics, law, economy and mass media?

This is complementary to LaclauGPT rather than a replacement for discourse analysis:

- **Laclau:** articulation, signifiers, antagonism, equivalence, hegemony.
- **Luhmann:** communication, system/environment distinction, functional differentiation, codes/programmes, structural coupling and cross-system observation/translation.

Avoid reducing Luhmann to a simplistic one-label classifier. The layer should model uncertainty and allow multiple/system references.

## Core theoretical sources

Start with at least:

- Niklas Luhmann, *Social Systems* (English trans. Stanford University Press, 1995; German original *Soziale Systeme*, 1984).
- Niklas Luhmann, *The Reality of the Mass Media* (Stanford University Press, English translation 2000).
- Niklas Luhmann, *Law as a Social System* (Oxford University Press, English ed. 2004).
- Niklas Luhmann, *Theory of Society*, Vols. 1–2 (Stanford University Press, English eds. 2012/2013).

Important implementation caution: do not invent a universal fixed list of binary codes. Codes/programmes differ by system and interpretation, and some source material may only weakly support assigning a functional system.

## Empirical / computational operationalization sources

This proposal should explicitly review work that tries to make Luhmann empirically/computationally usable:

1. **Loet Leydesdorff (1996), “Luhmann's Sociological Theory: Its Operationalization and Future Perspectives,” Social Science Information 35(2): 283–306.**
   - Operationalizes Luhmann-type reflexive communication using Shannon information theory.
   - DOI: https://doi.org/10.1177/053901896035002007

2. **Loet Leydesdorff (2011), “‘Meaning’ as a Sociological Concept: A Review of the Modeling, Mapping and Simulation of the Communication of Knowledge and Meaning,” Social Science Information 50(3–4): 391–413.**
   - Discusses semantic maps, simulation and empirical operationalization of communication/meaning.
   - DOI: https://doi.org/10.1177/0539018411411021

3. **Loet Leydesdorff, Alexander M. Petersen & Inga Ivanova (2017), “Self-organization of Meaning and the Reflexive Communication of Information,” Social Science Information 56(1): 4–27.**
   - Bridges Luhmannian meaning processing with information-theoretic empirical analysis.
   - DOI: https://doi.org/10.1177/0539018416675074

4. **“Network analysis as an alternative way to interpret constitutions” (PLOS ONE, 2021).**
   - Example of network analysis informed by Luhmannian legal-system distinctions.
   - https://doi.org/10.1371/journal.pone.0259461

5. Review newer empirical communication work using Luhmann, including the 2024 Frontiers article **“Communicating science addressing contentious environmental issues: utilizing Luhmann's social systems theory.”**
   - https://doi.org/10.3389/fcomm.2024.1348078

## Proposed data model

Add only optional derived fields/components, for example:

```text
systems_analysis:
  primary_system
  referenced_systems[]
  organization_context
  system_environment_distinction
  communication_code_candidates[]
  programme_or_criterion_candidates[]
  structural_couplings[]
  cross_system_translations[]
  observations_of_other_systems[]
  evidence_spans[]
  confidence
  model_method
  provenance
  human_validation
```

Potential system labels could include, where theoretically justified and codebook-defined:

- politics
- law
- science
- economy
- mass media
- education
- religion
- health/medicine
- art

Do not assume every social-media post cleanly belongs to one functional system. `unknown`, `mixed`, `organization-mediated`, and multiple references must be possible.

### 1. Codebook + structured extraction

Create a versioned Luhmann codebook with:

- definition of each supported system
- valid code/programme candidates
- positive/negative examples
- cross-system translation examples
- explicit uncertainty rules

Use Pydantic structured output for LLM-assisted extraction, retaining evidence spans and model provenance.

### 2. Classical NLP / embedding comparison

Benchmark LLM output against non-LLM alternatives:

- sentence-transformers embeddings + nearest prototype/class centroid
- supervised multilabel classifier with scikit-learn
- optional transformer classifier
- zero-shot NLI baseline

The layer should support human-coded training/evaluation samples rather than treating LLM classifications as ground truth.

### 3. Information-theoretic experiments

Inspired by Leydesdorff, add optional research scripts/notebooks for:

- Shannon entropy of distributions of communicative categories
- mutual information between source/actor/system/frame/signifier
- temporal change in entropy/information
- system × concept contingency matrices
- semantic-vector spaces / co-occurrence maps

Do **not** label these automatically as autopoiesis or self-organization without a defensible operational definition.

Candidate Python libraries:

- `numpy`, `scipy`
- `pandas` / `polars`
- `scikit-learn`
- `sentence-transformers`
- `statsmodels`
- `networkx` / `igraph`

### 4. Cross-system translation graph

Create derived graph relations such as:

```text
science --translated_as--> politics
science --cited_by--> media
economy --justifies--> politics
civil-society claim --juridified_as--> law
```

Every edge must retain source record IDs/evidence/provenance.

Useful network measures may include system-to-system flow counts, normalized transition matrices, brokerage and temporal change, but these are empirical graph measures, not inherently Luhmannian concepts.

### 5. Temporal analysis

Allow analysis of how an AI claim changes as it moves across contexts, e.g.:

```text
model capability result (science)
→ competitiveness claim (economy/politics)
→ regulatory-risk claim (politics/law)
→ headline/event framing (media)
```

This could be especially useful for AI26.

## Visualization outputs for Data Visualization

Expose optional derived tables/graph data for:

- system × system flow matrix
- Sankey/alluvial translation flows
- temporal system shares
- network of functional-system references
- system × signifier / system × imaginary heatmaps

Visualization should be downstream and not required for this issue to be considered successful.

## Evaluation

Before adopting as anything beyond experimental:

- build a small human-coded validation set
- report multilabel precision/recall/F1 where applicable
- report inter-annotator disagreement for theoretically ambiguous examples
- compare LLM vs embedding/classical baselines
- document categories that cannot be operationalized reliably

## Acceptance criteria for an experimental prototype

- [ ] theory/source review added
- [ ] empirical/computational Luhmann literature reviewed
- [ ] versioned codebook/schema designed
- [ ] optional structured extraction implemented or prototyped
- [ ] human-coded evaluation sample possible
- [ ] at least one non-LLM baseline implemented
- [ ] information-theoretic/semantic-map experiment documented
- [ ] optional cross-system graph representation designed
- [ ] full provenance/evidence retained
- [ ] feature disabled by default
- [ ] documentation clearly distinguishes Luhmannian interpretation from measured computational features
